### DISCUSSION TITLE
Dynamically change widget atributes

### DISCUSSION THREAD

**Discussion by ervinpopescu** [qtile/qtile#3548] [Q&A] [answered]
For example, foreground for `TextBox` / `GenPollText`(i'm trying to do a async. loop to change the color of a "battery widget", i know, i know, there is one already, just messing around), `disable_drag` for `GroupBox` (from a keybinding). What would be the correct way to do this?

**Comment by elParaguayo** [comment, marked as answer]
From a keybinding:
``` python
lazy.widget["groupbox"].eval("self.disable_drag=True")
```

You can also get the widget from the `qtile` object:
``` python
widget = qtile.widgets_map["groupbox"]
widget.disable_drag = True
```

**Reply by ervinpopescu** [reply]
Ohhh, forgot about eval. Thanks!

And the loop to change widget attributes based on some condition (the TextBox case)? Is that possible?

Right now, I'm using a @hook.subscribe.setgroup :)

**Reply by elParaguayo** [reply]
I didn't understand what you wanted a loop for. Hooks sound like a better approach.

**Reply by ervinpopescu** [reply]
Well, I wanted to check every other minute or so if the battery is low then change the foreground to red. You are right, a hook IS a better approach. Maybe there was a way of designing one so you could get "subscribed" to a widget's update interval? I will try and look into this on my own, but my knowledge of the codebase is limited to say the best :)

**Reply by elParaguayo** [reply]
You'd be better off creating a new widget which inherits from battery widget. You can then add a timer to that widget.

**Reply by m-col** [reply]
If that's all you want to do, you can use the `low_background` option, see http://docs.qtile.org/en/stable/manual/ref/widgets.html#libqtile.widget.Battery

**Reply by ervinpopescu** [reply]
> You'd be better off creating a new widget which inherits from battery widget. You can then add a timer to that 

Actually I don't use the battery widget at all. It's way too customizable for my needs. I guess I could just write a new widget... I've been wanting to do this for some time but never found the time.
